content/en/blog/_posts/2024-02-22-diy-create-your-own-cloud-with-kubernetes-part-1/index.md

```diff
@@ -0,0 +1,158 @@
+
+```
+cat config.yaml | docker run --rm -i -v /dev:/dev --privileged "ghcr.io/siderolabs/imager:v1.6.4" - 
+```
+
+And as a result, you will get an image with everythin you need, which you can use to install Talos Linux on your servers. This image will contain all the necessary firmware and kernel modules.
+
+But the question arises, how do you deliver the freshly formed image to your nodes?
+
+I have been contemplating the idea of PXE booting for quite some time. For example, the Kubefarm project that I [wrote article about](https://kubernetes.io/blog/2021/12/22/kubernetes-in-kubernetes-and-pxe-bootable-server-farm/) two years ago was entirely built on using this approach. But unfortunately, it does help you to deploy your very first parent cluster that will hold the others. So now we have prepared a simple solution that will help you do this the same using PXE approach.
+
+Essentially, all you need to do is run temporary DHCP and PXE servers in Docker containers. Then your nodes will boot from your image, and you can use a simple Debian-like script to help you bootstrap your nodes.
+
+[![asciicast](https://asciinema.org/a/627123.svg)](https://asciinema.org/a/627123)
+
+Source of this [talos-bootstrap](https://github.com/aenix-io/talos-bootstrap/) script available on GitHub
+
+This script allows you to deploy Kubernetes on bare metal in five minutes and obtain a kubeconfig for accessing it. However, many unresolved issues still lie ahead.
+
+### Delivering System Components
+
+At this stage, you already have a Kubernetes cluster capable of running various workloads. However, it is fully functional yet. In other words, you need to set up networking and storage, as well as install necessary cluster extensions, like KubeVirt to run virtual machines, as well the monitoring stack and other system-wide components.
+
+Traditionally, this is solved by installed Helm charts into your cluster. You can do this by running `helm install` commands locally, but this approach becomes inconvenient when you want track updates, and if you have multiple clusters and you want to keep them uniform. In fact, there are plenty of ways to do this declaratively. To solve this, I recommend using best GitOps practices. I mean tools like ArgoCD and FluxCD.
+
+While ArgoCD is more convenient for dev purposes with its graphical interface and a central control plane. FluxCD, on the other hand, is better suited for creating a Kubernetes distributions. With FluxCD, you can specify which charts with what parameters should be launched and describe dependencies. Then, FluxCD will take care of everything for you.
+
+It is suggested that performing a one-time installation of FluxCD in your newly created cluster and providing it with the configuration will cause it to install everything necessary, bringing the cluster to the expected state. For example, after installing Cozystack, you receive the following set of pre-installed Helm charts:
```

Comment:
```suggestion
It is suggested to perform a one-time installation of FluxCD in your newly created cluster and provide it with the configuration. This will install everything necessary, bringing the cluster to the expected state. 
```
Duplicate for example sentences, we can remove one.